paywall/src/__tests__/hooks/browser/useListenForPostMessage.test.js

```diff
@@ -28,164 +32,145 @@ describe('useListenForPostMessage hook', () => {
     config = { isServer: false, isInIframe: true }
   })
 
-  it('sets up the event listener', () => {
-    expect.assertions(1)
-
-    rtl.testHook(() => useListenForPostMessage(fakeWindow), {
-      wrapper,
+  function MockListener({
+    /* eslint-disable react/prop-types */
+    type = 'listenFor',
+    defaultValue = 'hi',
+    validator = false,
+    /* eslint-enable react/prop-types */
+  }) {
+    const value = useListenForPostMessage(fakeWindow, {
+      type,
+      defaultValue,
+      validator,
     })
+    return <div title="value">{value}</div>
+  }
 
-    expect(fakeWindow.addEventListener).toHaveBeenCalledWith(
-      'message',
-      expect.any(Function)
-    )
-  })
-
-  it('only subscribes on mount, and not on update', () => {
-    expect.assertions(1)
-
-    const { rerender } = rtl.testHook(
-      () => useListenForPostMessage(fakeWindow),
-      {
-        wrapper,
-      }
-    )
-
-    rerender()
-
-    expect(fakeWindow.addEventListener).toHaveBeenCalledTimes(1)
-  })
-
-  it('unsubscribes on unmount', () => {
-    expect.assertions(1)
-
-    const { unmount } = rtl.testHook(
-      () => useListenForPostMessage(fakeWindow),
-      {
-        wrapper,
-      }
-    )
-
-    unmount()
-
-    expect(fakeWindow.removeEventListener).toHaveBeenCalledWith(
-      'message',
-      expect.any(Function)
-    )
-  })
+  function getListener() {
+    return fakeWindow.addEventListener.mock.calls[0][1]
+  }
 
-  it('ignores calls on the server', () => {
-    expect.assertions(1)
+  describe('does nothing in invalid contexts', () => {
+    it.each([
```

Comment:
👍

Reply:
yeah isn't this cool? could definitely slim up some of the tests like CreatorLockForm validation